File: Programmers/K번째 수.js
// 정렬 > K번째 수
function solution(array, commands) {
    var answer = [];
    for (const command of commands) {
        let [i, j, k] = command
        let sortedArray = array.slice(i-1, j).sort((a, b) => a - b)
        answer.push(sortedArray[k-1])
    }
    return answer;
}

// for 루프 대신 map 을 사용
function solution2(array, commands) {
    var answer = [];
    commands.map((command) => {
            let [i, j, k] = command
            let sortedArray = array.slice(i-1, j).sort((a, b) => a - b)
            answer.push(sortedArray[k-1])
        }
    )
    return answer;
}

File: Programmers/Level 1/K번째수.py
def solution(array, commands):
    return list(map(lambda x: sorted(array[x[0]-1:x[1]])[x[2]-1], commands))

# def solution(array, commands):
#     answer = []
#     for command in commands:
#         x, y, z = command
#         temp = array[x-1:y]
#         temp.sort()
#         answer += [temp[z-1]]
#     return answer


# print(solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3], [4, 4, 1], [1, 7, 3]]))
